Use CPROVER_PREFIX macro instead of hardcoded __CPROVER_ string

tautschnig opened the pull request:
Using string literals is prone to typos and sometimes, but not always, using the CPROVER_PREFIX macro removes the potential for central modifications.

Fixes: #735

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] My contribution is formatted in line with CODING_STANDARD.md.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

tautschnig committed:
Use CPROVER_PREFIX macro instead of hardcoded __CPROVER_ string

Using string literals is prone to typos and sometimes, but not always, using the
CPROVER_PREFIX macro removes the potential for central modifications.

Fixes: #735

tautschnig committed:
Linter and clang-format ignores, and whitespace cleanup

thk123 commented on src/goto-programs/slice_global_inits.cpp:
```diff
@@ -43,7 +43,8 @@ void slice_global_inits(goto_modelt &goto_model)
   INVARIANT(
-    !directed_graph.empty(), "At least __CPROVER_start should be reachable");
+    !directed_graph.empty(),
+    "at least " CPROVER_PREFIX "start should be reachable");
```
:pick: `goto_functionst::entry_point()` inplace of it all if anything

tautschnig replied:
Good one, because also that one was always wrong: it was missing an extra underscore...

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 836af2c3f77dde7b451b64032bb1af8a8514ad5b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87932897

thk123 commented on src/cpp/cpp_internal_additions.cpp:
```diff
@@ -63,25 +63,28 @@ void cpp_internal_additions(std::ostream &out)
+  out << "unsigned long " CPROVER_PREFIX "next_thread_id = 0;" << '\n';
   out << "extern unsigned char "
-      << "__CPROVER_memory[__CPROVER::constant_infinity_uint];" << '\n';
+      << CPROVER_PREFIX "memory[__CPROVER::constant_infinity_uint];" << '\n';
```
:pick: Looks like there are a few more `__CPROVER` here?  perhaps not, since not underscore after, maybe there should be ?

tautschnig replied:
This is the `__CPROVER` namespace, without an underscore at the end. Arguably it could be debated whether that should change as well, but it isn't literally the same as all the rest...

thk123 reviewed (approved):
What a terrifying number of magic strings duplicated everywhere, this is certainly an improvement though.

chrisr-diffblue reviewed (approved):
Good to see this cleanup! Is it worth also adding a linter and/or clang-format rule to catch the use of raw magic __CPROVER_blah strings?

tautschnig committed:
Check for uses of __CPROVER_ in the linter

This should help avoid future code introducing literal uses of __CPROVER_ back
into the code base.

tautschnig commented:
@chrisr-diffblue Linter check added.

chrisr-diffblue commented:
Thanks for the additional linter check @tautschnig - definitely LGTM now.

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 21a5586cc6837a7e5103d6d5726784222a225cc2).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87942241

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: d942999a576e542ec82b62464b6fe6f7a58a863d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87943412

martin-cs reviewed (approved):
Looks good, thanks for the clean up.

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 4da3fa5f17b0403d5689b896114a346ffa2c0dc2).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88740531

allredj reviewed (commented):
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 3a67fb64aaa74144d4469fbbcc8fb750ba85333c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89554147
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.
